scripts/globals/pasteventwatcher.lua

```diff
@@ -0,0 +1,220 @@
+        [102] = -- La Theine Plateau
+        {
+            [1] = function() return (player:hasCompletedQuest(BASTOK, dsp.quest.id.bastok.DARK_PUPPET)) end,
+            [2] = function() return (player:hasCompletedQuest(WINDURST, dsp.quest.id.windurst.HITTING_THE_MARQUISATE)) end,
+            [3] = function() return (player:hasCompletedQuest(WINDURST, dsp.quest.id.windurst.I_CAN_HEAR_A_RAINBOW)) end,
+            [4] = function() return (player:hasCompletedQuest(SANDORIA, dsp.quest.id.sandoria.A_TIMELY_VISIT)) end,
+            [5] = function() return (player:hasCompletedQuest(SANDORIA, dsp.quest.id.sandoria.A_TIMELY_VISIT)) end,
+            [6] = function() return (player:hasCompletedMission(COP, dsp.mission.id.cop.THE_MOTHERCRYSTALS)) end,
+            [7] = function() return (player:hasCompletedMission(COP, dsp.mission.id.cop.THREE_PATHS)) end,
+            [8] = function() return (player:completeQuest(OTHER_AREAS_LOG, dsp.quest.id.otherAreas.WAKING_THE_BEAST)) end,
+            [9] = function() return (player:completeQuest(OTHER_AREAS_LOG, dsp.quest.id.otherAreas.WAKING_THE_BEAST)) end,
+            [10] = function() return (player:hasCompletedQuest(JEUNO, dsp.quest.id.jeuno.CHOCOBO_ON_THE_LOOSE)) end,
+            [11] = function() return (player:hasCompletedQuest(ABYSSEA, dsp.quest.id.abyssea.A_GOLDSTRUCK_GIGAS)) end,
+            [12] = function() return (player:hasCompletedQuest(ABYSSEA, dsp.quest.id.abyssea.A_GOLDSTRUCK_GIGAS)) end,
+            [13] = function() return (player:hasCompletedQuest(ABYSSEA, dsp.quest.id.abyssea.AN_OFFICER_AND_A_PIRATE)) end,
+            [14] = function() return (player:hasCompletedQuest(ABYSSEA, dsp.quest.id.abyssea.TENUOUS_EXISTENCE)) end,
+            [15] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [16] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [17] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [18] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [19] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [20] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [21] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [22] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [23] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [24] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [25] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+            [26] = false,               -- QUEST NOT IMPLEMENTED (ROV)
+
+        },
+        [106] = -- North Gustaberg
+        {
+            [1] = function() return (player:hasCompletedQuest(WINDURST, dsp.quest.id.windurst.AS_THICK_AS_THIEVES)) end,
+            [2] = function() return (player:hasCompletedQuest(BASTOK, dsp.quest.id.bastok.THE_GUSTABERG_TOUR)) end,
+            [3] = false, --function() return (player:hasCompletedQuest(APOCALIPSE RELIC WEAPON no way to track)) end,
```

Comment:
Could we not abuse `hasItem` for this check?

```lua
function () return player:hasItem(18306) end,
```

Granted, it could get cumbersome to check for all 10 potential versions from 75 to 119-3.

Reply:
I think thats the only way we can here is loop a hasItem through IDs. Maybe make a helper function  in utils because checking if players owns a version of a given relic will be needed elsewhere in the future as well I am almost sure. the quests for those aren't real quests on DSP and I doubt retail has a hidden quest entry that they just never display so its that or permanent playervars and player vars will make me grind you up and feed you to the chocobos..

Reply:
I am leaving it as false until someone finds a better way to implement a check if the char has ever completed the quest (since if it is trade from char to char the check for it would not be 100% accurate [like if he traded it from a mule but that mule didn't really do the quest])

Reply:
uh..you can't trade relics to your mule bro.

Reply:
thanks @TeoTwawki  made changes =) wasn't aware that relics couldn't be traded =)